src/actions/signage-actions.js

```diff
@@ -0,0 +1,444 @@
+ **/
+
+import {
+    getRequest,
+    createAction,
+    stopLoading,
+    startLoading,
+    authErrorHandler,
+    escapeFilterValue,
+    putRequest,
+    postRequest,
+    deleteRequest,
+    fetchErrorHandler
+} from 'openstack-uicore-foundation/lib/utils/actions';
+import {getAccessTokenSafely} from '../utils/methods';
+import Ably from "ably";
+
+export const REQUEST_SIGN = 'REQUEST_SIGN';
+export const RECEIVE_SIGN = 'RECEIVE_SIGN';
+export const UPDATE_SIGN = 'UPDATE_SIGN';
+export const SIGN_UPDATED = 'SIGN_UPDATED';
+export const SIGN_ADDED = 'SIGN_ADDED';
+export const REQUEST_SIGNAGE_EVENTS = 'REQUEST_SIGNAGE_EVENTS';
+export const RECEIVE_SIGNAGE_EVENTS = 'RECEIVE_SIGNAGE_EVENTS';
+export const RECEIVE_SIGNAGE_TEMPLATES = 'RECEIVE_SIGNAGE_TEMPLATES';
+export const REQUEST_SIGNAGE_BANNERS = 'REQUEST_SIGNAGE_BANNERS';
+export const RECEIVE_SIGNAGE_BANNERS = 'RECEIVE_SIGNAGE_BANNERS';
+export const REQUEST_SIGNAGE_LOCATIONS = 'REQUEST_SIGNAGE_LOCATIONS';
+export const RECEIVE_SIGNAGE_LOCATIONS = 'RECEIVE_SIGNAGE_LOCATIONS';
+export const UPDATE_SIGNAGE_BANNER = 'UPDATE_SIGNAGE_BANNER';
+export const SIGNAGE_BANNER_UPDATED = 'SIGNAGE_BANNER_UPDATED';
+export const SIGNAGE_BANNER_ADDED = 'SIGNAGE_BANNER_ADDED';
+export const SIGNAGE_BANNER_DELETED = 'SIGNAGE_BANNER_DELETED';
+export const SIGNAGE_UPDATED = 'SIGNAGE_UPDATED';
+export const SIGNAGE_STATIC_BANNER_UPDATED = 'SIGNAGE_STATIC_BANNER_UPDATED';
+
+const getAblyChannel = (summitId, locationId) => `SIGNAGE:${summitId}:${locationId}`;
+
+const realtimeAbly = new Ably.Realtime(process.env['ABLY_API_KEY']);
+
+const publishToAblyChannel = async (channel, key, message) => {
+    const ablyChannel = realtimeAbly.channels.get(channel);
+    
+    ablyChannel.subscribe((msg) => {
+        console.log("Received: " + JSON.stringify(msg.data));
+    });
+    
+    ablyChannel.publish(key, message);
+}
+
+export const getSign = (locationId) => async (dispatch, getState) => {
+    
+    const {currentSummitState} = getState();
+    const accessToken = await getAccessTokenSafely();
+    const {currentSummit} = currentSummitState;
+    
+    dispatch(startLoading());
+    
+    const params = {
+        access_token: accessToken,
+        'filter[]': [`location_id==${locationId}`]
+    };
+    
+    return getRequest(
+      createAction(REQUEST_SIGN),
+      createAction(RECEIVE_SIGN),
+      `${window.API_BASE_URL}/api/v1/summits/${currentSummit.id}/signs`,
+      authErrorHandler,
+      {}
+    )(params)(dispatch).then(() => {
+          dispatch(stopLoading());
+      }
+    );
+};
+
+export const getTemplates = () => async (dispatch, getState) => {
+    const {currentSummitState} = getState();
+    const accessToken = await getAccessTokenSafely();
+    const {currentSummit} = currentSummitState;
+    
+    const response = await fetch(`${window.SIGNAGE_BASE_URL}/templates.json?access_token=${accessToken}`);
+    if (response.ok) {
+        const {fntech} = await response.json();
```

Comment:
we need to be able to be multi tenant here
bc open infra foundation will have another set of templates per summit
so i do think that we need to define the tenant name per environment variable